src/Kunstmaan/MediaBundle/Twig/MediaTwigExtension.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Kunstmaan\MediaBundle\Twig;
+
+use Imagine\Filter\Basic\Crop;
+use Kunstmaan\MediaBundle\Entity\CroppableMediaLink;
+use Kunstmaan\MediaBundle\Helper\ManipulateImageService;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class MediaTwigExtension extends AbstractExtension
+{
+    /** @var ManipulateImageService */
+    private $manipulateImageService;
+
+    public function __construct(
+        ManipulateImageService $manipulateImageService
+    ) {
+        $this->manipulateImageService = $manipulateImageService;
+    }
+
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction(
+                'cropped_imagine_filter',
+                [$this, 'getCroppedImage']
+            ),
+            new TwigFunction(
+                'get_focus_point_class',
+                [$this, 'getFocusPointClass']
+            ),
+        ];
+    }
+
+    public function getCroppedImage(CroppableMediaLink $croppableMediaLink, string $view = '', string $filter = null)
+    {
+        if ($filter) {
```

Comment:
Why is this if needed? `$filter` can always be passed to the `cropImage` method, no?

Reply:
I wrote it so that when a developer explicitely provides an empty string to the twig function it won't cause the page to give a 500 error. by checking for empty here and allowing the default value of 'optim' to be used in the service method we can avoid that.